monai/transforms/composables.py

```diff
@@ -37,77 +38,130 @@
+                The original orientation, rotation, shearing are not preserved.
+
+                If False, the axes orientation, orthogonal rotation and
+                translations components from the original affine will be
+                preserved in the target affine. This option will not flip/swap
+                axes against the original ones.
+            mode (`reflect|constant|nearest|mirror|wrap`):
+                The mode parameter determines how the input array is extended beyond its boundaries.
+                Default is 'constant'.
+            cval (scalar): Value to fill past edges of input if mode is "constant". Default is 0.0.
             interp_order (int or sequence of ints): int: the same interpolation order
-                for all data indexed by `self,keys`; sequence of ints, should
+                for all data indexed by `self.keys`; sequence of ints, should
                 correspond to an interpolation order for each data item indexed
                 by `self.keys` respectively.
-            keep_shape (bool): whether to maintain the original spatial shape
-                after resampling. Defaults to False.
-            output_key (hashable): key to be added to the output dictionary to track
-                the pixdim status.
+            dtype (None or np.dtype): output array data type, defaults to None to use input data's dtype.
+            meta_key_format (str): key format to read/write affine matrices to the data dictionary.
         """
         MapTransform.__init__(self, keys)
-        self.affine_key = affine_key
-        self.spacing_transform = Spacing(pixdim, keep_shape=keep_shape)
+        self.spacing_transform = Spacing(pixdim, diagonal=diagonal, mode=mode, cval=cval, dtype=dtype)
         interp_order = ensure_tuple(interp_order)
         self.interp_order = interp_order \
             if len(interp_order) == len(self.keys) else interp_order * len(self.keys)
-        self.output_key = output_key
+        self.meta_key_format = meta_key_format
 
     def __call__(self, data):
         d = dict(data)
-        affine = d[self.affine_key]
-        original_pixdim, new_pixdim = None, None
         for key, interp in zip(self.keys, self.interp_order):
-            d[key], original_pixdim, new_pixdim = self.spacing_transform(d[key], affine, interp_order=interp)
-        d[self.output_key] = {'original_pixdim': original_pixdim, 'current_pixdim': new_pixdim}
+            affine_key = self.meta_key_format.format(key, 'affine')
+            original_key = self.meta_key_format.format(key, 'original_affine')
+            affine = d.get(affine_key, None)
+            if affine is None:
+                affine = d.get(original_key, None)
+            # resample array of each corresponding key
+            # using affine fetched from d[affine_key]
+            d[key], affine, new_affine = self.spacing_transform(
```

Comment:
Suggest using 'affine_' as your orientation code and no need to copy?

Reply:
thanks, now fixed